### ISSUE TITLE
Write integration tests

### ISSUE THREAD

**Issue by ggwadera** [garageScript/c0d3-app#670]
Since a lot of production breaking bugs have been happening recently, writing integration tests to assure full API/client functionality can help in reduce the number of these bugs.

Basic implementation steps:

1. Start Server
2. Send API request
3. Save response as a snapshot

This will probably be better implemented using a in-memory embedded database like SQLite to avoid connecting to an external database, the recent Prisma seed PR #659  can help in seeding this db with fake data for the tests.

**Comment by tomrule007** [comment]
I'm glad to see progress is getting made on this issue. Last week I had mentioned how switching to the `getLayout` pattern interfered with our current tests and I thought [next-page-tester](https://www.npmjs.com/package/next-page-tester) might be a good solution. After reading this little blog post https://dev.to/toomuchdesign/dom-testing-next-js-applications-46ke  it sounds like the real solution is to switch to intergration tests like this and possible impliment some e2e tests as well.

**Comment by ggwadera** [comment]
Yeah some e2e tests running the entire Next.js stack might be good too, so we can test the SSG and SSR rendering with Next's data fetching and also the API layer. But for the front-end components I believe most cases can be covered by the unit tests.

With E2E tests we could test an entire flow of our app's logic, for example: account creation -> login -> send submissions -> receive review -> complete lesson. That might help in detecting some regressions like we already had before.

**Comment by songz** [comment]
Sample PR for reference: https://github.com/garageScript/c0d3-app/pull/992
